pytorch_lightning/trainer/trainer.py

```diff
@@ -2479,6 +2480,33 @@ def _active_loop(self) -> Optional[Union[FitLoop, EvaluationLoop, PredictionLoop
     Logging properties
     """
 
+    @property
+    def logger(self) -> Optional[LightningLoggerBase]:
+        return self._logger
+
+    @logger.setter
+    def logger(self, new_logger: Optional[LightningLoggerBase]) -> None:
+        self._logger = new_logger
+        if not new_logger:
+            self._loggers = []
+        elif isinstance(new_logger, LoggerCollection):
+            self._loggers = list(new_logger)
+        else:
+            self._loggers = [new_logger]
+
+    @property
+    def loggers(self) -> List[LightningLoggerBase]:
+        return self._loggers
+
+    @loggers.setter
+    def loggers(self, new_loggers: Optional[Iterable[LightningLoggerBase]]) -> None:
```

Comment:
```suggestion
    def loggers(self, new_loggers: Optional[List[LightningLoggerBase]]) -> None:
```

Reply:
if we made this not optional then we could make the change below

Reply:
Iterable is the annotation used here

https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/trainer/trainer.py#L134

Reply:
See https://github.com/PyTorchLightning/pytorch-lightning/pull/11683#discussion_r798804419 as well.